src/repr/src/proto/mod.rs

```diff
@@ -139,3 +142,23 @@ impl<T: ProtoRepr> ProtoRepr for Option<T> {
         repr.map(T::from_proto).transpose()
     }
 }
+
+/// Convenience syntax for trying to convert a `Self` value of type
+/// `Option<U>` to `T` if the value is `Some(value)`, or returning
+/// [`TryFromProtoError::MissingField`] if the value is `None`.
+pub trait TryIntoIfSome<T> {
+    fn try_into_if_some<S: ToString>(self, field: S) -> Result<T, TryFromProtoError>;
+}
+
+/// A blanket implementation for `Option<U>` where `U` is the
+/// `Proto$T` type for `T`.
+impl<T, U> TryIntoIfSome<T> for Option<U>
+where
+    T: TryFrom<U, Error = TryFromProtoError>,
+{
+    fn try_into_if_some<S: ToString>(self, field: S) -> Result<T, TryFromProtoError> {
+        self.map(T::try_from)
+            .transpose()?
+            .ok_or_else(|| TryFromProtoError::missing_field(field))
```

Comment:
I *think* this can be simplified (not tested):

```
self.ok_or_else(|| TryFromProtoError::missing_field(field))?
    .try_into()
```

Gets rid of the `transpose` which not everyone know what it does.